packages/components/src/components/drawer/drawer.jsx

```diff
@@ -15,54 +15,45 @@ const IconDrawer = ({ className }) => (
     </svg>
 );
 
-class Drawer extends React.PureComponent {
-    constructor(props) {
-        super(props);
-        this.state = { is_open: props.is_open };
-    }
+const Drawer = ({
+    anchor = 'left',
+    children,
+    className,
+    is_open,
+    toggleDrawer,
+    contentClassName,
+    footer,
+    header,
+    width = 250,
+    zIndex = 4,
+}) => {
+    const [will_open, set_is_open] = React.useState(is_open);
 
-    componentWillReceiveProps(nextProps) {
-        this.setState({
-            is_open: nextProps.is_open,
-        });
-    }
+    React.useEffect(() => {
+        set_is_open(is_open);
+    }, [is_open]);
 
-    toggleDrawer = () => {
-        this.setState(
-            prev_state => ({
-                is_open: !prev_state.is_open,
-            }),
-            () => {
-                if (this.props.toggleDrawer) {
-                    this.props.toggleDrawer(this.state.is_open);
-                }
+    const toggle_drawer = () => {
+        set_is_open(!will_open);
```

Comment:
```suggestion
        set_is_open(!will_open);
```

So when u use `useStateCallback` u can pass this function as the second argument of the setState

```
() => {
    if (toggleDrawer) {
        toggleDrawer(will_open);
     } 
```

This will mimic the old behaviour of this code where a function needs to be called after state is set. You can omit the `useEffect` if you use this method

Reply:
done